src/Bicep.Core.UnitTests/Assertions/ErrorBuilderAssertions.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace Bicep.Core.UnitTests.Assertions
+{
+    public static class ErrorBuilderExtensions
+    {
+        public static ErrorBuilderAssertions Should(this DiagnosticBuilder.ErrorBuilderDelegate errorBuilder) => new(errorBuilder);
+    }
+
+    public class ErrorBuilderAssertions : ReferenceTypeAssertions<DiagnosticBuilder.ErrorBuilderDelegate, ErrorBuilderAssertions>
```

Comment:
>ErrorBuilderAssertions [](http://example.com/codeflow?start=17&length=22)

This class is basically identical to the Diagnostic* one. Not sure if there's a way to reuse the logic somehow, but it's also not likely to change.

Reply:
It might be different as Error Diagnostic will never have warning level, therefore some methods might be simplified.

Reply:
Haha yeah, that's a good pt. Can't have errors with warning severity.

Reply:
Will removed the `level` parameter.